Cesium3DTilesSelection/include/Cesium3DTilesSelection/TileContentFactory.h

```diff
@@ -92,13 +93,20 @@ class CESIUM3DTILESSELECTION_API TileContentFactory final {
    * If a matching loader is found, it will be applied to the given
    * input, and the result will be returned.
    *
+   * @param asyncSystem The async system to use during the loading.
+   * @param pAssetAccessor The asset accessor to use if needed.
+   * @param requestHeaders The request headers to be used if needed.
    * @param input The {@link TileContentLoadInput}.
    * @return The {@link TileContentLoadResult}, or `nullptr` if there is
    * no loader registered for the magic header of the given
    * input, and no loader for the content type of the input.
    */
-  static std::unique_ptr<TileContentLoadResult>
-  createContent(const TileContentLoadInput& input);
+  static CesiumAsync::Future<std::unique_ptr<TileContentLoadResult>>
+  createContent(
+      const CesiumAsync::AsyncSystem& asyncSystem,
+      const std::shared_ptr<CesiumAsync::IAssetAccessor>& pAssetAccessor,
+      const std::vector<std::pair<std::string, std::string>>& requestHeaders,
```

Comment:
I was thinking about that too, but I was a little worried about needless copies and the overhead of the atomic reference count increment with the shared_ptrs. For instance: 
- Should the AsyncSystem be wrapped with a shared_ptr in TilesetExternals, or should AsyncSystem be kept as a reference in TileContentLoadInput?
- I assume the pAssetAccessor can be a shared ptr without issue.
- How would I avoid making a copy of the request headers? Could I even move it into TileContentLoadInput? I assumed not since the Tile may fail at loading and try again, possibly needing the request headers again.

Reply:
> Should the AsyncSystem be wrapped with a shared_ptr in TilesetExternals, or should AsyncSystem be kept as a reference in TileContentLoadInput?

AsyncSystem should be stored by value in TileContentLoadInput.

> I assume the pAssetAccessor can be a shared ptr without issue.

Yep

> How would I avoid making a copy of the request headers? Could I even move it into TileContentLoadInput? I assumed not since the Tile may fail at loading and try again, possibly needing the request headers again.

The `data` is already passed by reference (via a span) to TileContentLoadInput, so if that is ok then it should be ok to do the same with the headers. But actually that's a little worrying. I don't think anything is currently guaranteeing that the lifetime of that data (or headers) is extended to the end of the Future returned by the loader. So if the loader continues to access the data/headers after the initial load function returns and the `std::shared_ptr<IAssetRequest>` passed to the continuation goes out of scope, then it'll be undefined behavior.

An easy and efficient - but conceptually messy - solution here is to std::move the entire `std::shared_ptr<IAssetRequest>` into the `TileContentLoadInput`. Then we can remove `data`, `contentType`, and `url`, and we don't need to add `headers`. That'll save some allocating/copying. It's messy because ideally content loading wouldn't assume that the content always came from an asset request. Given that IAssetRequest is polymorphic, though, it's not totally disastrous. For example, we could easily write an `IAssetRequest` that represents an entirely in-memory asset. So I think that might be the best solution, barring a better idea that we haven't thought of yet. :)